spec/models/ability_spec.rb

```diff
@@ -89,11 +89,104 @@
+          let(:registration_for_user_with_open_registration_and_limit_excedeed) { create(:registration, conference: conference_with_open_registration_and_limit_excedeed, user: user2) }
+
+          it{ should_not be_able_to(:new, Registration.new(conference_id: registration_for_user_with_open_registration_and_limit_excedeed.conference_id)) }
+          it{ should_not be_able_to(:create, Registration.new(conference_id: registration_for_user_with_open_registration_and_limit_excedeed.conference_id)) }
+        end
+      end
+
+      context 'when user is a speaker with event confirmed' do
+        let(:conference_with_speaker_confirmed) { create(:conference) }
+        let(:event_with_speaker_confirmed) { create(:event, state: 'confirmed', program: conference_with_speaker_confirmed.program) }
+
+        before :each do
+          event_with_speaker_confirmed.speakers << user
+        end
+
+        context 'when speaker has already registered' do
+
+          before :each do
+            create(:registration, conference: conference_with_speaker_confirmed, user: user2)
+          end
+
+          it{ should_not be_able_to(:new, Registration.new(conference: conference_with_speaker_confirmed)) }
+          it{ should_not be_able_to(:create, Registration.new(conference: conference_with_speaker_confirmed)) }
+        end
+
+        context 'when the registration period is not set' do
+          it{ should_not be_able_to(:new, Registration.new(conference: conference_with_speaker_confirmed)) }
+          it{ should_not be_able_to(:create, Registration.new(conference: conference_with_speaker_confirmed)) }
+        end
+
+        context 'And the registration period is set' do
+          let(:registration_period) { create(:registration_period) }
+
+          before :each do
+            conference_with_speaker_confirmed.registration_period = registration_period
+          end
+
+          context 'with registration open' do
+
+            it{ should be_able_to(:new, Registration.new(conference: conference_with_speaker_confirmed)) }
+            it{ should be_able_to(:create, Registration.new(conference: conference_with_speaker_confirmed)) }
+
+            context 'and registration_limit_exceeded' do
+
+              before :each do
+                conference_with_speaker_confirmed.registration_limit = 1
+                create(:registration, conference: conference_with_speaker_confirmed, user: user2)
+              end
+
+              it{ should be_able_to(:new, Registration.new(conference: conference_with_speaker_confirmed)) }
+              it{ should be_able_to(:create, Registration.new(conference: conference_with_speaker_confirmed)) }
+            end
+          end
+
+          context 'with registration closed' do
+            let(:event_with_speaker_confirmed) { create(:event, state: 'confirmed', program: conference_with_closed_registration.program) }
```

Comment:
Where is it defined that this conference has a closed registration?

Reply:
check this https://github.com/openSUSE/osem/blob/master/spec/models/ability_spec.rb#L33